C++ API compatibility, again

MisterDA opened the pull request:
https://manpages.debian.org/unstable/gcc-15-diagnostics/sarif-replay-15.1.en.html
```sh
docker build -t sarif-replay-15 - <(cat <<'EOF'
# syntax=docker/dockerfile:1
FROM debian:unstable
RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
    --mount=type=cache,target=/var/lib/apt,sharing=locked \
    apt update && apt-get --no-install-recommends install -y \
    gcc-15-diagnostics
ENTRYPOINT ["/usr/bin/sarif-replay-15"]
EOF
)
docker run --rm -t sarif-replay-15 --usage

source="$PWD/report.sarif"
docker run --rm --mount "type=bind,source=$source,target=/tmp/sarif" \
           -t sarif-replay-15 /tmp/sarif
```

kit-ty-kate commented on testsuite/tests/cxx-api/gcc-13.sh:
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
```
```suggestion
set -e
```

MisterDA committed:
ocamltest: expose an has-cxx action, cxx and ccomp_type variables

MisterDA committed:
ocamltest: retry deleting response files

In some runs on Windows, ocamltest is unable to delete response
files. Perhaps all processes haven't yet closed them. Try deleting
them 3 times. If it still fails, mark the files to be deleted on next
reboot (Windows doesn't have a tmpfs).

MisterDA committed:
configure: check for a C++ compiler

AC_PROG_CXX has to occur before LT_INIT to prevent this error:

    warning: AC_REQUIRE: `AC_PROG_CXX' was expanded before it was required
    https://www.gnu.org/software/autoconf/manual/autoconf.html#Expanded-Before-Required

It seems to be impossible (or too difficult) to enable this check
conditionally, based on whether ocamltest was enabled or not.

MisterDA committed:
unixsupport.h: fix C++ _Atomic incompatibility

The `std::` namespace was missing from
bdbe619f1655a02c3ce18dcb745a4cf96c672dc2.

MisterDA committed:
ocamltest: contain temporary files in a single directory

MisterDA committed:
Test for C++ API compatibility GCC and MSVC SARIF diagnostics

MisterDA committed:
Update Changes

MisterDA committed:
Revert "Skip sockaddr_cxx.ml in Jenkins"

This reverts commit 17a207b5ff49761eddc7c9f6ce76b11543f07a9c.

MisterDA committed:
Revert "ocamltest: retry deleting response files"

This reverts commit 1258af4f64fc5c62667311ad40f1ba72ab94e783.
